### PR TITLE
containerd containers can use host network now

### PR DESCRIPTION
**Pull request by h4l0gen**
# containerd containers can use network now

containerd container can use host network connection for trace_sni test.

## How to use

Run test, with containerd, and containers can connect to network now. 

## Testing done

YES
fixes: #3184

### CODE REVIEW

**Comment by h4l0gen** [comment]
Hi @mauriciovasquezbernal, main changes:
- Here I decided to always use the host network for containerd containers in the test environment.
- in pkg/container-utils/testutils/containerd.go:
  Modified the Run method to always use the host network
  
  I was also thinking of using `Bridge Network` or `CNI` approach instead of using host's network directly. Please @mauriciovasquezbernal provide your thoughts on this. Thank you!

**Comment by h4l0gen** [comment]
image

**Comment by mauriciovasquezbernal** [comment]
> I was also thinking of using Bridge Network or CNI approach instead of using host's network directly. Please @mauriciovasquezbernal provide your thoughts on this. Thank you!

It'd be much better if it uses a different namespace. Do you think it's easy to implement?

File: gadgets/trace_sni/test/trace_sni_test.go [hunk 1]
```diff
@@ -69,7 +65,7 @@ func TestTraceSNI(t *testing.T) {
 	testContainer := containerFactory.NewContainer(
 		containerName,
 		"while true; do setuidgid 1000:1111 wget --no-check-certificate -T 2 -q -O /dev/null https://inspektor-gadget.io; sleep 0.1; done",
-		containerOpts...,
+		containers.WithContainerImage(containerImage),
```
Comment (mauriciovasquezbernal) [inline gadgets/trace_sni/test/trace_sni_test.go:68]: Are all changes related to image really needed?

Comment (h4l0gen) [reply gadgets/trace_sni/test/trace_sni_test.go:68]: Hi @mauriciovasquezbernal, yes you are right, actually they don't require, I just remembered I was doing some debugging with that and forgot to change it back.

**Comment by h4l0gen** [comment]
Hi @mauriciovasquezbernal , I've tried a lot with `netns` package. But not able to capture event from container. Its something that, namespace is not setting up correctly for container. Its really getting complicated. Additionally, I found this [blog](https://medium.com/@teddyking/namespaces-in-go-network-fdcf63e76100)  
To use this:
- Compile the netsetgo.go file and place the binary in /usr/local/bin/.
- Set the correct permissions: sudo chown root:root /usr/local/bin/netsetgo && sudo chmod 4755 /usr/local/bin/netsetgo

I don't know, it would suit with IG tool or not. What you think about it?

Also 
> It'd be much better if it uses a different namespace. Do you think it's easy to implement?

Yes, its little complicated, but if you really want using different namespace approach, I will try more to solve it.

**Comment by mauriciovasquezbernal** [comment]
> Hi @mauriciovasquezbernal , I've tried a lot with `netns` package. But not able to capture event from container. Its something that, namespace is not setting up correctly for container. Its really getting complicated. Additionally, I found this [blog](https://medium.com/@teddyking/namespaces-in-go-network-fdcf63e76100) To use this:
> 
>     * Compile the netsetgo.go file and place the binary in /usr/local/bin/.
> 
>     * Set the correct permissions: sudo chown root:root /usr/local/bin/netsetgo && sudo chmod 4755 /usr/local/bin/netsetgo
> 
> 
> I don't know, it would suit with IG tool or not. What you think about it?

I'm sorry, I can't follow your message. Could you please try to explain (1) what you are trying to do and (2) the problems you're facing?

**Pull request closed without merging**